test/e2e/aro.go

```diff
@@ -18,62 +18,59 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/azureclient"
 )
 
-const (
-	wSsh = "99-worker-ssh"
-	mSsh = "99-master-ssh"
-)
-
-// The high level idea of these tests is to definitively say that a cluster is an ARO cluster.
-var _ = Describe("Verify Attributes of ARO Cluster", func() {
-	ctx := context.Background()
-	// acrDomainList should contain acrDomain verifier
-	acrDomainList := []string{"arointsvc.azurecr.io", "arointsvc.azurecr.us", "arosvc.azurecr.io", "arosvc.azurecr.us"}
-	azEnvironmentList := []string{azureclient.PublicCloud.Environment.Name, azureclient.USGovernmentCloud.Environment.Name}
-	It("should be possible to definitively confirm cluster is ARO", func() {
-		// Get cluster object
-		co, err := clients.AROClusters.AroV1alpha1().Clusters().Get(context.Background(), "cluster", metav1.GetOptions{})
+// The high level idea of these tests is to definitively say that
+// a cluster is an ARO cluster. This should help us catch regressions
+// when migrating between installers (e.g. vendored one vs CLI one run by Hive)
+var _ = Describe("ARO Cluster", func() {
+	It("must have ARO-specific machine configs", func() {
+		ctx := context.Background()
+
+		By("listing machine config pools")
+		mcs, err := clients.MachineConfig.MachineconfigurationV1().MachineConfigs().List(ctx, metav1.ListOptions{})
 		Expect(err).NotTo(HaveOccurred())
 
-		mcp, err := clients.MachineConfig.MachineconfigurationV1().MachineConfigPools().List(ctx, metav1.ListOptions{})
-		Expect(err).NotTo(HaveOccurred())
-
-		// Expect 99-worker-ssh and 99-master-ssh to exist
-		masterSsh, workerSsh := false, false
-		for _, m := range mcp.Items {
-			for _, mc := range m.Spec.Configuration.Source {
-				if mc.Name == wSsh {
-					workerSsh = true
-				}
-				if mc.Name == mSsh {
-					masterSsh = true
-				}
-			}
+		By("verifying that ARO-specific machine configs exist")
+		actualMachineConfigNames := []string{}
+		for _, mc := range mcs.Items {
+			actualMachineConfigNames = append(actualMachineConfigNames, mc.Name)
 		}
-		By("Checking if workerSsh and masterSsh is enabled or disabled")
-		Expect(workerSsh).To(BeTrue())
-		Expect(masterSsh).To(BeTrue())
+		Expect(actualMachineConfigNames).To(ContainElements([]string{
+			"90-aro-worker-registries",
+			"99-master-aro-dns",
+			"99-master-fips",
```

Comment:
`fips` isn't guaranteed, it's keyed off of the cluster document -> `properties.clusterProfile.fipsValidatedModules == "Enabled"`

Can we put in a check to include the `99-master|worker-fips` machineconfigs iff the property is enabled in the clusterProfile?

Reply:
Technically we're creating all E2E clusters currently with FIPS enabled, but if we change that, someone will come back and discover this.  :smile: